feature-libs/product-configurator/rulebased/core/state/effects/configurator-basic-effect.service.ts

```diff
@@ -0,0 +1,57 @@
+import { Injectable } from '@angular/core';
+import { Configurator } from '../../model/configurator.model';
+
+/**
+ * Service that provides helper methods for the basic configurator effects,
+ * in order to enhance them without the need to introduce new effects
+ */
+@Injectable({ providedIn: 'root' })
+export class ConfiguratorBasicEffectService {
+  /**
+   * Finds first group with attributes for a configuration. Throws error if such a group does not exist,
+   * as this is an illegal state
+   * @param configuration
+   * @returns Group id
+   */
+  getGroupWithAttributesForConfiguration(
```

Comment:
I would suggest to rename the method to `getGroupId(...)`

Reply:
Agree